examples/dashboard/src/components/ServerSdkContext.tsx

```diff
@@ -26,33 +32,77 @@ type Props = {
 };
 
 export const ServerSDKProvider = ({ children }: Props) => {
-  const [serverAddress, setServerAddressState] = useState<string>(() => {
-    const serverAddress = localStorage.getItem(localStorageServerAddress);
-    return serverAddress ? serverAddress : "localhost:4000";
-  });
-
-  const setServerAddress = useCallback(
-    (value: string) => {
-      setServerAddressState(value);
-      localStorage.setItem(localStorageServerAddress, value);
-    },
-    [setServerAddressState]
+  const [host, setHost] = useLocalStorageStateString(LOCAL_STORAGE_HOST_KEY, "localhost:5002");
+  const [protocol, setProtocol] = useLocalStorageStateString(LOCAL_STORAGE_PROTOCOL_KEY, "wss");
+  const [path, setPath] = useLocalStorageStateString(LOCAL_STORAGE_PATH_KEY, "/socket/peer/websocket");
+  const [isSecure, setIsSecure] = useLocalStorageStateString(LOCAL_STORAGE_PROTOCOL_KEY, "wss");
+
+  const [signalingWebsocket, setSignalingWebsocket] = useState<string | null>(null);
+  const [serverMessagesWebsocket, setServerMessagesWebsocket] = useState<string | null>(null);
+  const [httpApiUrl, setHttpApiUrl] = useState<string | null>(null);
+
+  const [serverToken, setServerToken] = useLocalStorageStateString("serverToken", "development");
+
+  const setHostInput = useCallback((value: string) => {
+    setHost(value);
+    localStorage.setItem(LOCAL_STORAGE_HOST_KEY, value);
+  }, []);
+
+  const setProtocolInput = useCallback((value: string) => {
+    setProtocol(value);
+    localStorage.setItem(LOCAL_STORAGE_PROTOCOL_KEY, value);
+  }, []);
+
+  const setPathInput = useCallback((value: string) => {
+    setPath(value);
+    localStorage.setItem(LOCAL_STORAGE_PATH_KEY, value);
+  }, []);
+
+  useEffect(() => {
+    const restProtocol = protocol === "wss" ? "https" : "http";
+
+    const abc = `${restProtocol}://${host}`;
+    console.log({ abc });
+    setHttpApiUrl(abc);
+  }, [host, protocol]);
```

Comment:
I believe that the `httpApiUrl` does not need to be calculated inside `useEffect`. This can easily be turned into `useMemo`. We can even turn this into a simple variable, as concatenation of two strings shouldn't be expensive.

Reply:
simple variable is nice 👍